src/condor_dagman/dagman_submit.cpp

```diff
@@ -607,26 +318,25 @@ direct_condor_submit(const Dagman &dm, Job* node,
 	// Otherwise we'll initialize and parse it in from the submit file later
 	SubmitHash* submitHash = node->GetSubmitDesc();
 
-	TmpDir		tmpDir;
-	std::string	errMsg;
-	const char* directory = node->GetDirectory();
-	if (!tmpDir.Cd2TmpDir(directory, errMsg)) {
-		debug_printf(DEBUG_QUIET,
-			"Could not change to node directory %s: %s\n",
-			directory, errMsg.c_str());
-		return false;
-	}
 	int rval = 0;
 	bool is_factory = param_boolean("SUBMIT_FACTORY_JOBS_BY_DEFAULT", false);
 	bool success = false;
 	std::string errmsg;
 	Qmgr_connection * qmgr = NULL;
 	auto_free_ptr owner(my_username());
-	char * qline = NULL;
-	const char * queue_args = NULL;
+	char * qline = nullptr;
+	const char * queue_args = nullptr;
 	MacroStreamFile ms;
 	DCSchedd schedd;
 
+	auto vars = init_vars(dm, *node);
+
+	auto sort_prepend = [](NodeVar lhs, NodeVar rhs) -> bool {
+		if (lhs.append == rhs.append) { return false; }
+		return ! lhs.append;
+	};
+	std::stable_sort(vars.begin(), vars.end(), sort_prepend);
```

Comment:
Do you really want `std::stable_partition` here?  (or `std::ranges::stable_partition`